contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -3950,28 +3959,31 @@ load_categories_hash(RawStmt *cats_sql, MemoryContext per_query_ctx)
 					 errmsg("provided \"categories\" SQL must " \
 							"return 1 column of at least one row")));
 
-		for (i = 0; i < proc; i++)
+		Assert(spi_tuptable->numvals >= tuple_processed);
+		for (i = 0; i < tuple_processed; i++)
 		{
 			bbf_pivot_cat_desc *catdesc;
 			char	   *catname;
+			char	   *catname_lower;
 			HeapTuple	spi_tuple;
 
 			/* get the next sql result tuple */
 			spi_tuple = spi_tuptable->vals[i];
 
 			/* get the category from the current sql result tuple */
 			catname = SPI_getvalue(spi_tuple, spi_tupdesc, 1);
-			if (catname == NULL)
+			catname_lower = downcase_identifier(catname, strlen(catname), false, false);
+			if (catname_lower == NULL)
 				ereport(ERROR,
 						(errcode(ERRCODE_SYNTAX_ERROR),
 						 errmsg("provided \"categories\" SQL must " \
 								"not return NULL values")));
 			
 			SPIcontext = MemoryContextSwitchTo(per_query_ctx);
```

Comment:
Pls use oldMemoryContext for the var name in here, the SPIcontext has other meaning for var naming.

Reply:
Renamed in latest commit.